connect/runtime/src/main/java/org/apache/kafka/connect/util/KafkaBasedLog.java

```diff
@@ -163,6 +163,18 @@ public void start() {
         log.info("Started KafkaBasedLog for topic " + topic);
     }
 
+    /**
+     * Sleep for some time so that topic used for this KafkaBasedLog gets created. Note that
+     * {@code System.currentTimeMillis()} is not monotonic, so check for that condition.
```

Comment:
Why don't we use a monotonic timer?

Reply:
Wanted to make least amount of change. I can update the code to use monotonic 'nanoTime` instead (nanoseconds in Time interface). We will also need to convert that to milli before passing to sleep (unless we want to add nano to those interfaces too, like Utils.sleep)

Reply:
In fact I don't like this loop altogether. Going to rewrite it so that it doesn't use these constructs.

Reply:
You can use `hiResClockMs` so that you get the value in `ms`.

Reply:
In fact we don't need value from clock to sleep, only need it to find elapsed time and timeout. I have decoupled these two, and also fixed a minor issue where it was sleeping first time even if topic was present.

Please take a look.